pkg/volume/volume.go

```diff
@@ -223,95 +222,133 @@ func (err deletedVolumeInUseError) Error() string {
 	return string(err)
 }
 
-func RenameDirectory(oldPath, newName string) (string, error) {
-	newPath, err := ioutil.TempDir(filepath.Dir(oldPath), newName)
+// RenameWithFallback attempts to rename a file or directory, but falls back to
+// copying in the event of a cross-link device error. If the fallback copy
+// succeeds, src is still removed, emulating normal rename behavior.
+func RenameWithFallback(src, dest string) error {
+	fi, err := os.Lstat(src)
 	if err != nil {
-		return "", err
+		return err
 	}
 
-	// os.Rename call fails on windows (https://github.com/golang/go/issues/14527)
-	// Replacing with copyFolder to the newPath and deleting the oldPath directory
-	if runtime.GOOS == "windows" {
-		err = copyFolder(oldPath, newPath)
-		if err != nil {
-			glog.Errorf("Error copying folder from: %s to: %s with error: %v", oldPath, newPath, err)
-			return "", err
+	// If renaming a directory make sure the destination does not exist.
+	if fi.IsDir() {
```

Comment:
I think we should either return an error or just force it to be unique through this process.  Removing it is a little scary, and sort of defeats the purpose of this function which was "remove a dir, but make it appear to have been removed as quickly as possible".  The rename is the "quickly as possible" part - callers see it disappear, but the actual delete can be slow.

Blocking on this delete, aside from being scary, is slow :)